### PR TITLE
Implement S+ and SS+ Grading

### PR DESCRIPTION
**Pull request by frenzibyte**
### Implement SH and XH (S+ and SS+) grades when playing locally.
NOTE: I've discovered an issue in this pull request, the value of the grade is automatically changed if the accuracy was changed in ScoreProcessor, and it's bound to the break's grade value, so that'll cause an issue of displaying the grade not adjusted in a break period.
Preview

### CODE REVIEW

**Comment by iiSaLMaN** [comment]
I'm thinking about simplifying the rank adjusting way, instead of calling a function to adjust the rank from completion function, how about we simply create a field that tells the grading function whether to adjust the rank or not, and will be set from ApplyToScoreProcessor

File: osu.Game/Rulesets/Mods/IApplicableToScoreProcessor.cs [hunk 1]
```diff
@@ -11,5 +12,6 @@ namespace osu.Game.Rulesets.Mods
     public interface IApplicableToScoreProcessor : IApplicableMod
     {
         void ApplyToScoreProcessor(ScoreProcessor scoreProcessor);
+        ScoreRank AdjustRank(ScoreRank rank);
```
Comment (peppy) [inline osu.Game/Rulesets/Mods/IApplicableToScoreProcessor.cs]: missing newline, missing xmldoc.

File: osu.Game.Tests/Visual/Gameplay/TestCasePlayerLoader.cs [hunk 2]
```diff
@@ -110,6 +111,11 @@ public void ApplyToScoreProcessor(ScoreProcessor scoreProcessor)
             {
                 Applied = true;
             }
+
+            public ScoreRank AdjustRank(ScoreRank rank)
```
Comment (peppy) [inline osu.Game.Tests/Visual/Gameplay/TestCasePlayerLoader.cs]: `=> rank;`

File: osu.Game/Rulesets/Mods/ModFlashlight.cs [hunk 3]
```diff
@@ -46,6 +47,11 @@ public void ApplyToScoreProcessor(ScoreProcessor scoreProcessor)
             Combo.BindTo(scoreProcessor.Combo);
         }
 
+        public ScoreRank AdjustRank(ScoreRank rank)
```
Comment (peppy) [inline osu.Game/Rulesets/Mods/ModFlashlight.cs]: `=> rank;`

**Review by peppy — CHANGES_REQUESTED** [review]

**Comment by iiSaLMaN** [comment]
I guess i'm gonna close this and create another branch for implementing the grades and make sure it will not contain any issues

**Pull request closed without merging**